src/renderer/components/dock/dock.store.ts

```diff
@@ -165,18 +169,21 @@ export class DockStore implements DockStorageState {
   }
 
   @action
-  createTab(anonTab: IDockTab, addNumber = true): IDockTab {
-    const tabId = MD5(Math.random().toString() + Date.now()).toString();
-    const tab: IDockTab = { id: tabId, ...anonTab };
+  createTab({ id = uuid.v4(), ...rawTab }: IDockTab, { addNumber = false, open = true } = {}): IDockTab {
+    const tab = { id, ...rawTab };
 
     if (addNumber) {
       const tabNumber = this.getNewTabNumber(tab.kind);
 
       if (tabNumber > 1) tab.title += ` (${tabNumber})`;
     }
+
     this.tabs.push(tab);
-    this.selectTab(tab.id);
-    this.open();
+
+    if (open) {
```

Comment:
How this changes related to PR's scope btw?

Reply:
I guess they are not, will remove.